### DISCUSSION TITLE
Can't bind std::vector<T> with nanobind/stl/vector.h include present

### DISCUSSION THREAD

**Discussion by Lorac** [wjakob/nanobind#128] [Q&A] [answered]
I'm on the `f80e73402e0ab00fb73ce6e9ab7c975942c83c94` ref, so currently master. GCC 11.3 with C++20.

When trying to bind a `nb::bind_vector<std::vector<int16_t>>(m, "VectorShort");`  or any kind of std::vector

```
site-packages/nanobind/include/nanobind/nb_class.h:207:40: error: static assertion failed: Attempted to create a constructor for a type that won't be handled by the nanobind's class type caster. Is it possible that you forgot to add NB_MAKE_OPAQUE() somewhere?
  207 |             detail::make_caster<Type>::IsClass,
```

I'm presented with this error. I should not use the NB_MAKE_OPAQUE when reading the documentation and example.

From testing, it looks like we can't combine having the include of stl/vector and stl/bind_vector, was this intended? This snippet causes the compiler to fail.

```c++
#include <nanobind/stl/vector.h> // Need it for something else
#include <nanobind/stl/bind_vector.h>

NB_MODULE(test_bind_vector_ext, m) {
    nb::bind_vector<std::vector<unsigned int>>(m, "VectorInt");
}
```

**Comment by wjakob** [comment, marked as answer]
Hi @Lorac  -- I added an FAQ entry to answer this question, please let me know anything remains unclear: https://nanobind.readthedocs.io/en/latest/faq.html#compilation-fails-with-a-static-assertion-mentioning-nb-make-opaque
